windows/src/global/delphi/general/Keyman.System.CanonicalLanguageCodeUtils.pas

```diff
@@ -47,22 +51,22 @@ class function TCanonicalLanguageCodeUtils.FindBestTag(const Tag: string; AddReg
       t.Tag := Result;
     end;
 
-    if not TLangTagsMap.LangTags.TryGetValue(t.Language, LangTag) then
+    if not TLangTagsMap.LangTags.TryGetValue(t.Tag, LangTag) then
     begin
-       // Not a valid language subtag but perhaps it's a custom language
-       // We'll make no further assumptions
+      // Not a known tag but perhaps it's a custom language
+      // We'll make no further assumptions
       Exit(t.Tag);
     end;
 
     // Then, lookup the lang-script and see if there is a suppress-script
-    if SameText(t.Script, LangTag.script) and LangTag.suppress then
-      t.Script := ''
     // Or add the default script in if it is missing and not a suppress-script
-    else if (t.Script = '') and not LangTag.suppress then
+    if (t.Script = '') and not LangTag.suppress and AddScriptIfNotSuppressed then
+      // AddScriptIfNotSuppressed will generally be True for Windows scenarios;
+      // for other systems and for registry systems it will be False
```

Comment:
When would Windows use False? Is it a specific list of tags?

Reply:
Windows never uses false :)

Reply:
Turns out Linux also uses `True` (https://github.com/keymanapp/keyman/pull/4439/files#diff-9d2b72bb38a2d410351c312d1a57d8a2f410572f21f45a8adad1a2ef1fe549abR265)